startup/94-multi_spectra.py

```diff
@@ -0,0 +1,677 @@
+        the str 'all' will result in all keys from the two dictionaries being
+        used provided they both have exactly the same keys.
+
+    Initialization Parameters
+    -------------------------
+    name : str
+        The name of the instantiated version of this class.
+    default_parameters_filepath : str or Path
+        The default filepath for the spectrum 'parameters' file.
+    default_settings_filepath : str or Path
+        The default filepath for the spectrum 'settings' file.
+    parameters_index_name : str
+        The column name that indicates the 'index' in the parameters file,
+        default is 'edge_name'.
+    settings_index_name : str
+        The column name that indicates the 'index' in the settings file,
+        default is 'edge_name'.
+
+    Attributes
+    ----------
+    parameters : FileDict
+        A FileDict object with an attribute dictionary that maps spectrum names
+        (like 'C1s') to the parameters that define the spectrum, which are by
+        definition 'low_energy', 'high_energy' and 'step_size'.
+    settings : FileDict
+        A FileDict object with an attribute dictionary that maps 'items' to a
+        dictionary mapping the ``ophyd.Device``'s that need to be set for each
+        'item' to the value to set it to. Examples might include items like
+        'det1.gain' or 'det1.exposure_time', but can include any 'settable'
+        ``ophyd.Device`` object.
+    function : func
+        A function that is to be called at step 2 above.
+    validate : method
+        A method that validates that a proposed 'item' can be used with the
+        currently loaded dictionaries.
+    '''
+
+    def __init__(self, name, function,
+                 default_parameters_filepath, parameters_index_name,
+                 default_settings_filepath, settings_index_name):
+        self.name = name
+        self.function = function
+        self.parameters = FileDict('parameters', default_parameters_filepath,
+                                   parameters_index_name)
+        self.settings = FileDict('settings', default_settings_filepath,
+                                 settings_index_name)
+
+    def __call__(self, items):
+        # check that spectra is a list or a str, if a str convert to a list
+        if items == 'all':
+            items = list(self.settings.keys())
+        elif isinstance(items, str):
+            items = [items]
+        elif not isinstance(items, list):
+            raise FileDataRouterValueError(
+                f'The items passed to ``{self.name}(items)`` is expected to be'
+                f' a str or a list, instead we got {items} which is of type '
+                f'{type(items)}.')
+
+        # reload the files to make sure any saved chagnes are found
+        self.settings.load()
+        self.parameters.load()
+
+        # check that items are keys in the two dictionaries
+        if (not set(items) <= set(self.parameters.dictionary)
```

Comment:
Can this condition be rewritten as `if (set(items) > set(self.parameters.dictionary)`? Less typing, easier to understand.
```suggestion
        if (set(items) > set(self.parameters.dictionary)
```